recipes/install.rb

```diff
@@ -115,10 +115,34 @@
   only_if { node["td_agent"]["includes"] }
 end
 
-package "td-agent" do
+# Fetch the source package if selected.
+if node["td_agent"]["install_url"]
+  remote_name = ::File.basename(
+    ::URI.parse(node["td_agent"]["install_url"]).path
+  )
+  remote_file 'td-agent install source' do
```

Comment:
Don't you have your own apt repository? I think that modifying the cookbook to allow using custom apt repository must be the most straightforward way to install custom packages. I feel this might be kinda adhoc solution only for your use case...

Reply:
We don't have enough packages to justify running our own repo.

We would keep using the official td-agent repo but until https://github.com/treasure-data/chef-td-agent/issues/87 is fixed this is the only option to ensure systems are automatically rebuilt without changes or interventions